client/login/wp-login/index.jsx

```diff
@@ -575,39 +595,82 @@ export class Login extends Component {
 			</svg>
 		);
 
+		const mainContent = (
+			<Main
+				className={ clsx( 'wp-login__main', {
+					'is-wpcom-migration': isFromMigrationPlugin,
+					'is-social-first': isSocialFirst,
+					'is-generic-oauth': isGenericOauth,
+				} ) }
+			>
+				{ this.renderI18nSuggestions() }
+
+				<DocumentHead
```

Comment:
It feels super weird to have this DocumentHead inside the Main component but I'm guessing removing it would require a much larger refactor 😅

Reply:
Because JSX is supposed to emulate markup, and in HTML you'd never have the document head inside the main content. The code may work correctly, but it feels horribly unsemantic.

Reply:
Not trying to speak on behalf of Luis, but I feel the question may have been "why is it a large refactor?".

Reply:
Oh! I noticed we're using `DocumentHead` like this in many places, so if we decided to move it somewhere where it makes more sense semantically we should probably be consistent and do it for all instances.

Reply:
For what is worth, Next.js also has a [`Head`](https://nextjs.org/docs/pages/api-reference/components/head) component that does exactly the same thing: set something in the `<head>` element.

Doing it with the proper nesting and "semantically" is probably outright impossible, because React apps never control the entire HTML document, they are always mounted in some sub-element of `<body>`.

Reply:
React 19 does [support all the metadata tags](https://react.dev/blog/2024/12/05/react-19#support-for-metadata-tags) dropped anywhere in your component tree, so hopefully at some point we'll be able to migrate to it 🤞